playwithllm/server
# Migration from CommitStreams to PlayWithLLM Platform

## Overview
This PR represents a major architectural restructuring and rebranding from CommitStreams to PlayWithLLM, implementing a microservices architecture with Docker support and message queue integration.

## Key Changes

### 1. Architecture Restructuring
- Implemented microservices architecture with three main services:
  - Authentication Service
  - Business Service
  - Inference Service
- Added Docker support with individual Dockerfiles for each service
- Integrated RabbitMQ for inter-service communication
- Added Redis configuration for caching
- Restructured project directories for better organization

### 2. Service-Specific Changes

#### Authentication Service
- Moved user, role, and resource management to dedicated admin domain
- Updated authentication strategies (Google, GitHub, Local)
- Improved authentication middleware with development bypass option

#### Business Service
- Added inference domain with CRUD operations
- Implemented RabbitMQ integration for inference requests
- Updated routing and middleware configuration

#### Inference Service
- Added new service for handling ML model inference
- Implemented queue workers for processing inference requests
- Added result handling and response management

### 3. Infrastructure Updates
- Added Docker Compose configuration with services for:
  - MongoDB
  - RabbitMQ
  - Redis
- Updated MongoDB connection string to use `pwllmdb`
- Added volume management for persistent data

### 4. Code Organization
- Moved shared components to `/shared` directory:
  - Configurations
  - Libraries
  - Middlewares
  - Database utilities
  - Error handling
- Updated import paths across the project
- Improved error handling and logging

### 5. Cleanup
- Removed GitHub-specific features (pull requests, repositories)
- Removed unused domains and schemas
- Updated branding from CommitStreams to PlayWithLLM

## Testing Instructions
1. Build and run services using Docker Compose:
```bash
docker-compose up --build
```
2. Verify each service starts correctly
3. Test authentication flows
4. Verify inference request processing
5. Check inter-service communication

## Notes
- Environment variables need to be configured as per `.env.example`
- MongoDB data will persist in the configured Docker volume
- Development mode includes authentication bypass option
